### PR TITLE
build: Upgrade puppeteer and related dependencies

### PR DESCRIPTION
**Pull request by tylerbutler**
wip

### CODE REVIEW

**Comment by jason-ha** [comment]
>       '@fluidframework/datastore-definitions': 2.0.0-internal.8.0.0

Okay with these changes?

---
Refers to: pnpm-lock.yaml:15730 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
> 

Probably shouldn't have all three of these.

---
Refers to: pnpm-lock.yaml:16602 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
>       sha.js: 2.4.11

Another roguish update

---
Refers to: pnpm-lock.yaml:17468 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
>       sha.js: 2.4.11

Nevermind, looks like good collateral update.

---
In reply to: [1930747138](https://github.com/microsoft/FluidFramework/pull/19508#issuecomment-1930747138) [](http://example.com/codeflow?ancestors=1930747138)

---
Refers to: pnpm-lock.yaml:17468 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
> 

Lingering internal reference?

---
Refers to: pnpm-lock.yaml:22953 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
> 

Should dedupe

---
Refers to: pnpm-lock.yaml:28902 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
>   /for-each/0.3.3:

probably should dedupe

---
Refers to: pnpm-lock.yaml:30373 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
>     dev: true

huh?

---
Refers to: pnpm-lock.yaml:32102 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by tylerbutler** [comment]
> > ```
> >   '@fluidframework/datastore-definitions': 2.0.0-internal.8.0.0
> > ```
> 
> Okay with these changes?
> 
> Refers to: pnpm-lock.yaml:15730 in [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157). [](commit_id = [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157), deletion_comment = False)

Tough to see what you're referring to in the GitHub UI, but I do expect some internal8.0.0 entries in the lockfile since that's what the typetest "previous version" is right now.

**Comment by tylerbutler** [comment]
>       '@fluidframework/datastore-definitions': 2.0.0-internal.8.0.0

Installed codeflow so I could see what you were referring to. I don't know how to keep pnpm from doing this. I can manually change them back to 8.0.0, but I don't know why pnpm changed themn in the first place.

---
In reply to: [1930739069](https://github.com/microsoft/FluidFramework/pull/19508#issuecomment-1930739069) [](http://example.com/codeflow?ancestors=1930739069)

---
Refers to: pnpm-lock.yaml:15730 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by tylerbutler** [comment]
> 

Do you mean running pnpm dedupe or something else? pnpm dedupe produces a lot of churn.

---
In reply to: [1930753177](https://github.com/microsoft/FluidFramework/pull/19508#issuecomment-1930753177) [](http://example.com/codeflow?ancestors=1930753177)

---
Refers to: pnpm-lock.yaml:28902 in 53eb47e. [](commit_id = 53eb47eca76fbd73a250107cb995acdb03e94157, deletion_comment = False)

**Comment by jason-ha** [comment]
> > ```
> >   '@fluidframework/datastore-definitions': 2.0.0-internal.8.0.0
> > ```
> 
> Installed codeflow so I could see what you were referring to. I don't know how to keep pnpm from doing this. I can manually change them back to 8.0.0, but I don't know why pnpm changed themn in the first place.
> 
> In reply to: [1930739069](https://github.com/microsoft/FluidFramework/pull/19508#issuecomment-1930739069)
> 
> Refers to: pnpm-lock.yaml:15730 in [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157). [](commit_id = [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157), deletion_comment = False)

Sorry, I thought the UI would be okay so long as I stuck to whole lines. I guess it can only be a single line.
I wanted to reference multiple lines in places.
The is a lot of duplication and maybe also inconsistent references to some packages like there are three versions of @fluidframework/core-interfaces.

**Comment by tylerbutler** [comment]
Regarding the lockfile changes, I reset the lockfile to main, nuked node-modules, and ran `pnpm i` again. Is there something else I should do to minimize lockfile churn?

**Comment by jason-ha** [comment]
> > 
> 
> Do you mean running pnpm dedupe or something else? pnpm dedupe produces a lot of churn.
> 
> In reply to: [1930753177](https://github.com/microsoft/FluidFramework/pull/19508#issuecomment-1930753177)
> 
> Refers to: pnpm-lock.yaml:28902 in [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157). [](commit_id = [53eb47e](https://github.com/microsoft/FluidFramework/commit/53eb47eca76fbd73a250107cb995acdb03e94157), deletion_comment = False)

If there was going to be a full dedupe, my experience is that it should be done apart from any other change. You just can't figure out what is going on with dedupe and other changes at the same time unless dedupe is done with every dep change.
So I meant less aggressive which is perhaps manual.

File: examples/utils/bundle-size-tests/package.json [hunk 1]
```diff
@@ -48,10 +48,11 @@
 		"@fluidframework/eslint-config-fluid": "^3.3.0",
 		"@mixer/webpack-bundle-compare": "^0.1.0",
 		"@types/node": "^18.19.0",
-		"@types/puppeteer": "1.3.0",
+		"@types/puppeteer": "7.0.4",
 		"cross-env": "^7.0.3",
```
Comment (jason-ha) [inline examples/utils/bundle-size-tests/package.json:52]: Needed?

File: examples/apps/attributable-map/tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
 		"module": "esnext",
 		"outDir": "lib",
 		"jsx": "react",
-		"types": ["jest", "puppeteer", "jest-environment-puppeteer", "expect-puppeteer"],
+		"types": ["jest", "puppeteer", "jest-environment-puppeteer"],
```
Comment (jason-ha) [inline examples/apps/attributable-map/tsconfig.json:7]: "puppeteer" can also be removed from this list now, right?

**Review by jason-ha — COMMENTED** [review]
I see two things. Remove examples/utils/bundle-size-tests @types/puppeteer reference and if possible the "puppeteer" in tsconfig "types" lines.

File: examples/apps/attributable-map/tsconfig.json (hunk 2 above)
Comment (jason-ha) [reply examples/apps/attributable-map/tsconfig.json]: The answer is "sometimes". Tests without using the types impacted won't need it, but good to have for consistent, 
 common setup.

**Pull request closed without merging**